src/lib/client.ts

```diff
@@ -105,117 +104,75 @@ function clearStored() {
   }
 }
 
-export const Client = async () => {
-  const wsProvider = new WsProvider()
-  const api = await ApiPromise.create({ provider: wsProvider, types: customTypes })
-  console.log(api.genesisHash.toHex())
-  console.log('init client...')
-  let unsubscribe: VoidFn = () => { }
-  // clearStored()
-  let farmed: FarmedBlock[] = getStoredBlocks()
-  let clientData = reactive(emptyData)
-  let clearTauriDestroy: event.UnlistenFn = () => { }
-  clientData.farming.farmed = farmed
-
-  const client = <ClientType>{
-    api,
-    data: clientData,
-    getStatus: {
-      async farming() { },
-      async plot() { },
-      async network() { }
-    },
-    do: {
-      unsubscribe,
-      blockSubscription: {
-        unsubscribe,
-        processSubscription: {},
-        clearStored,
-        clearTauriDestroy,
-        stopOnReload(this: any, ev: Event) {
-          ev.preventDefault()
-          client.do?.blockSubscription.stop()
-        },
-        async start() {
-          if (!client.api) throw (Error(`Api Missing, can't start block subscription yet.`))
-
-
-
-          this.unsubscribe = await client.api.rpc.chain.subscribeNewHeads(async (lastHeader) => {
-            const signedBlock = await api.rpc.chain.getBlock(lastHeader.hash)
-            for (const log of signedBlock.block.header.digest.logs) {
-              if (log.isPreRuntime) {
-                const [type, data] = log.asPreRuntime
-                if (type.toString() === 'POC_') {
-                  const poCPreDigest: PoCPreDigest = api.registry.createType('PoCPreDigest', data)
-                  const solution: Solution = api.registry.createType('Solution', poCPreDigest.solution)
-                  const farmerId: FarmerId = api.registry.createType('FarmerId', solution.public_key)
-                  console.log('farmerId: ', farmerId.toString())
-                }
+export class Client {
+  wsProvider = new WsProvider
+  api: ApiPromise = new ApiPromise
```

Comment:
I believe this will still not properly work until `Client.init()` is called. Please put a big TODO to refactor this is a way that doesn't have this issue.

And another TODO on top of this class to add visibility to all methods/properties, most of this stuff shouldn't be public.